src/Panel/ChangesDialog.php

```diff
@@ -2,70 +2,53 @@
 
 namespace Kirby\Panel;
 
-use Kirby\Cms\App;
-use Kirby\Cms\Find;
-use Kirby\Http\Uri;
-use Kirby\Toolkit\Escape;
-use Throwable;
+use Kirby\Cms\Collection;
+use Kirby\Content\Changes;
 
 class ChangesDialog
 {
-	public function changes(array $ids = []): array
-	{
-		$kirby     = App::instance();
-		$multilang = $kirby->multilang();
-		$changes   = [];
+	protected Changes $changes;
 
-		foreach ($ids as $id) {
-			try {
-				// parse the given ID to extract
-				// the path and an optional query
-				$uri   = new Uri($id);
-				$path  = $uri->path()->toString();
-				$query = $uri->query();
-				$model = Find::parent($path);
-				$item  = $model->panel()->dropdownOption();
+	public function __construct()
+	{
+		$this->changes = new Changes();
+	}
 
-				// add the language to each option, if it is included in the query
-				// of the given ID and the language actually exists
-				if (
-					$multilang &&
-					$query->language &&
-					$language = $kirby->language($query->language)
-				) {
-					$item['text'] .= ' (' . $language->code() . ')';
-					$item['link'] .= '?language=' . $language->code();
-				}
+	public function files(): array
+	{
+		return $this->items($this->changes->files());
+	}
 
-				$item['text'] = Escape::html($item['text']);
+	public function items(Collection $models): array
+	{
+		$items = [];
 
-				$changes[] = $item;
-			} catch (Throwable) {
-				continue;
-			}
+		foreach ($models as $model) {
+			$items[] = $model->panel()->dropdownOption();
 		}
 
-		return $changes;
+		return $items;
 	}
```

Comment:
```php
public function items(Collection $models): array
{
	return $models->values(
		fn ($model) => $model->panel()->dropdownOption()
	);
}	
```

Though maybe it doesn't need the `::items()` method even, if we can do simply the following?


```php
public function pages(): array
{
	return $this->changes->pages()->values(
		fn ($page) => $page->panel()->dropdownOption()
	);
}	
```

Reply:
This would be totally enough for now. But we would still need the items method later if we want to add the feature ideas from above. We could go for the simpler version first though. What do you prefer?

Reply:
Fine with me, then let's keep the `::items()`method but with `->values()`?